packages/dds/sequence/src/test/generateSharedStrings.ts

```diff
@@ -6,11 +6,12 @@
 import { SnapshotLegacy as Snapshot } from "@fluidframework/merge-tree";
 import Random from "random-js";
 import * as mocks from "@fluidframework/test-runtime-utils";
-import { SharedString } from "../sharedString";
-import { SharedStringFactory } from "../sequenceFactory";
+import { SharedString, TestSharedString } from "../sharedString";
+import { SharedStringFactory, TestSharedStringFactory } from "../sequenceFactory";
 import { IntervalType } from "../intervalCollection";
 
 export const LocationBase: string = "src/test/snapshots/";
+export const TestLocationBase: string = "src/test/snapshots/OLD/";
```

Comment:
"old" is relative. Better to use something like "v1intervals"

Reply:
Should the snapshot files with the v1 intervals be in a directory on the same level as legacy, legacyWithCatchUp, and v1? Or should there be a dedicated directory for v1 interval collections with legacy, legacyWithCatchUp, and v1 folders within the dedicated directory?

Reply:
I think ideally we'd just generate the test strings slightly differently. In the "before" diff the 3 versions refer to changes relevant to merge-tree, where here your changes add some back-compat for changes in interval collection (and doesn't particularly care about the merge-tree part of the snapshot). So rather than add a new "version" to the versions array we should probably have some structure more like 
```
for (const mergeTreeVersion of mergeTreeFormatVersions) {
  // yield some strings that generate interesting cases w.r.t merge-tree
}

for (const intervalCollectionVersion of intervalCollectionVersions) {
  // yield some strings that generate interesting cases for interval collections
}
```
and where we write the files is kind of immaterial as long as they don't conflict. It's not really a big deal though, PR as-is is an improvement so I'll approve :)